opnode/l2/source.go

```diff
@@ -157,5 +175,53 @@ func (s *Source) L2BlockRefByHash(ctx context.Context, l2Hash common.Hash) (eth.
 		// w%: wrap the error, we still need to detect if a canonical block is not found, a.k.a. end of chain.
 		return eth.L2BlockRef{}, fmt.Errorf("failed to determine block-hash of height %v, could not get header: %w", l2Hash, err)
 	}
-	return derive.BlockReferences(block, s.genesis)
+	return blockToBlockRef(block)
+}
+
+func blockToBlockRef(block *types.Block) (eth.L2BlockRef, error) {
+	txs := block.Transactions()
+	if len(txs) == 0 {
+		return eth.L2BlockRef{}, fmt.Errorf("l2 block is missing L1 info deposit tx, block hash: %s", block.Hash())
```

Comment:
This won't work on the genesis block. That's why the rollup genesis was always being passed around

Reply:
oh right, forgot about that, good catch